test-cypress/integration/help/rest-api.spec.js

```diff
@@ -0,0 +1,110 @@
+describe('Help / REST API', () => {
+
+    before(() => {
+        // Disables the security (only for this spec) to be able to interact with Swagger's iframe
+        // See https://github.com/cypress-io/cypress/issues/136
+        Cypress.config({chromeWebSecurity: false});
+    });
+
+    beforeEach(() => {
+        cy.visit('/webapi');
+        getSwaggerFrame()
+            .find('#message-bar')
+            .should('be.visible');
+    });
+
+    // TODO: Check explicitly for all controllers (name + summary) ?
+
+    it('should provide documentation and examples for GraphDB APIs', () => {
+        // The GraphDB API section should list the available controllers & each controller should have a heading and possible options
+        verifyControllerList(getGraphDBAPI());
+
+        // All controllers should be collapsed
+        verifyControllerListIsCollapsed(getGraphDBAPI());
+
+        // Examine the import controller
+        verifyControllerOptions('import-controller', 'Data import');
+    });
+
+    it('should provide documentation and examples for RDF4J APIs', () => {
+        // The RDF4J API section should list the available controllers & each controller should have a heading and possible options
+        verifyControllerList(getRDF4JAPI());
+
+        // All controllers should be collapsed
+        verifyControllerListIsCollapsed(getRDF4JAPI());
+
+        // Examine the import controller
+        verifyControllerOptions('repositories', 'Repository management');
+    });
+
+    function getSwaggerFrame() {
+        return cy.get('#webapi_frame').iframe();
+    }
+
+    function getGraphDBAPI() {
+        return getSwaggerFrame().find('#swagger-ui-container-graphdb');
+    }
+
+    function getRDF4JAPI() {
+        return getSwaggerFrame().find('#swagger-ui-container-rdf4j');
+    }
+
+    function getGraphDBController(id) {
+        return getGraphDBAPI().find('.resource .heading').contains(id).parentsUntil('#resources').last();
+    }
+
+    function verifyControllerList(controllerSection) {
+        controllerSection.within(($section) => {
+            cy.wrap($section)
+                .should('be.visible')
+                .find('.resource')
+                .should('be.visible')
+                .its('length')
+                .should('be.gt', 0);
+
+            cy.get('.resource .heading')
```

Comment:
Can't this be chained with the above?

Reply:
Maybe with `root().find()` ? Will try it

Reply:
Added .root() although I don't really see the benefit here because it will make a selection nevertheless, `cy.root().find() === cy.get()`

Reply:
I also updated a bit the chains in `verifyControllerOptions`

Reply:
> Added .root() although I don't really see the benefit here because it will make a selection nevertheless, `cy.root().find() === cy.get()`

But why? Isn't the result yeld by the above part, before the root actually `.result` list? Why do you need to return to the section and execute the selection for `.result > .heading` again? 
IMO you could just make `.find('.heading')...` instead of `root()`

Reply:
Cause there is `.its()` before that and `.find()` would not work in the chain.